webapp/channels/src/components/suggestion/at_mention_provider/at_mention_provider.tsx

```diff
@@ -377,32 +378,42 @@ export default class AtMentionProvider extends Provider {
             return false;
         }
 
-        if (this.lastCompletedWord && captured[0].trim().startsWith(this.lastCompletedWord.trim())) {
-            // It appears we're still matching a channel handle that we already completed
+        // Extract the actual matched text including the trigger character from the ORIGINAL pretext
+        // Find the @ or ＠ position within the matched text
+        const fullMatchIndex = captured.index;
+        const fullMatch = captured[0];
+        const atSymbolMatch = fullMatch.match(/[@＠]/);
+        const atSymbolOffset = atSymbolMatch ? fullMatch.indexOf(atSymbolMatch[0]) : 0;
+        const matchedPretextStart = fullMatchIndex + atSymbolOffset;
+        const matchedPretext = pretext.substring(matchedPretextStart);
```

Comment:
I'm going to take a look to see if I can simplify this a bit by tweaking the regex to include the at-sign instead of this string manipulation